### ISSUE TITLE
TypeError: string argument expected, got 'bytes'

### ISSUE THREAD

**Issue by shulhi** [xhtml2pdf/xhtml2pdf#278]
Code to generate the pdf
```
result = io.StringIO()
pdf = pisa.pisaDocument(io.StringIO(html), dest=result, link_callback=fetch_resources)
```

I'm using `xhtml2pdf (0.1a1)` and `reportlab (3.3.0)` with Python 3.4

Traceback:

```
[2016-03-22 16:04:02,455: INFO/MainProcess] Received task: reports.tasks.run_report[929c444a-ee7b-4b54-943a-b1a2b7bb36fb]
[2016-03-22 16:04:02,781: ERROR/MainProcess] Task reports.tasks.run_report[929c444a-ee7b-4b54-943a-b1a2b7bb36fb] raised unexpected: TypeError("string argument expected, got 'bytes'",)
Traceback (most recent call last):
  File "/Users/shulhi/.pyenv/versions/3.4.3/envs/ronda-webapi/lib/python3.4/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/Users/shulhi/.pyenv/versions/3.4.3/envs/ronda-webapi/lib/python3.4/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/Users/shulhi/Development/projects/ronda-webapi/web/reports/tasks.py", line 15, in run_report
    site = report.run(**options)
  File "/Users/shulhi/Development/projects/ronda-webapi/web/reports/types/attendance.py", line 27, in run
    self.make_pdf(ctx)
  File "/Users/shulhi/Development/projects/ronda-webapi/web/reports/types/attendance.py", line 40, in make_pdf
    pdf = pisa.pisaDocument(io.StringIO(html), dest=result, link_callback=fetch_resources)
  File "/Users/shulhi/.pyenv/versions/3.4.3/envs/ronda-webapi/lib/python3.4/site-packages/xhtml2pdf/document.py", line 184, in pisaDocument
    context.dest.write(data)  # TODO: context.dest is a tempfile as well...
TypeError: string argument expected, got 'bytes'
```

**Comment by shulhi** [comment]
The issue is with `data` is `bytes`, it needs to be `data.decode("ISO-8859-1")` when passing it into `context.dest.write`

**Comment by benjaoming** [comment]
Hi @shulhi - thanks for your report - #280 was merged and released in 0.1a2, could you check if this resolves it?

I think we want as little encoding and decoding as possible so we always have data passed around in the simplest format possible. But if your suggestion solves the issue, I'm happy to take a look at a PR, merge it, and release another one.
